service/sharddistributor/handler/handler.go

```diff
@@ -76,5 +79,31 @@ func (h *handlerImpl) Health(ctx context.Context) (*types.HealthStatus, error) {
 func (h *handlerImpl) GetShardOwner(ctx context.Context, request *types.GetShardOwnerRequest) (resp *types.GetShardOwnerResponse, retError error) {
 	defer func() { log.CapturePanic(recover(), h.logger, &retError) }()
 
-	return nil, nil
+	var owner string
+	switch request.GetNamespace() {
+	case constants.HistoryNamespace:
```

Comment:
I'd avoid having such switch/if statements in shard distributor code. The namespaces it manages should be config driven. e.g.
```
namespaces:
  history:
    type: udg-based-sharding # or something along these lines
   # other fields it needs to initialize the relevant inner components
```

Reply:
Hey. We will have to use this as a migration strategy for our existing distribution. In the future, both of these namespaces will be config-driven. Still, the onboarding step will be manual since Ii is not only udg-based but also has a specific hash function that we use in historyPeerResolver. I doubt that we can express the whole thing based on the config.

Reply:
Ok good to know you guys are also thinking of making this config driven. I understand it will take some time to decide on config structure. Since you want to quickly test whether shard manager can mimic existing behavior I'm fine with taking this short cut knowing that we will rewrite a config driven shard distributor shortly after validation.